app/controllers/posts_controller.rb

```diff
@@ -1,9 +1,10 @@
 class PostsController < ApplicationController
-  def show
-    @post = Post.find(params[:id])
+  def index
+    @users = [User.find(params[:user_id])]
+    @posts = @users[0].posts
   end
 
-  def index
-    @posts = Post.all
+  def show
+    @posts = [Post.find(params[:id])]
```

Comment:
- **[OPTIONAL]** There is no use in wrapping the queries on lines 3 and 8 within [...], You can simply use remove the outer [...] and in the views have `@user`, instead of `@users[0]`